location/detector.go

```diff
@@ -1,43 +1,40 @@
 package location
 
 import (
-	"errors"
-	"github.com/oschwald/geoip2-golang"
-	"net"
+	"github.com/mysterium/node/ip"
 )
 
 type detector struct {
-	databasePath string
+	ipResolver ip.Resolver
+	locationResolver Resolver
 }
 
-// NewDetector returns Detector which uses country database
-func NewDetector(databasePath string) *detector {
+// NewDetector constructs Detector
+func NewDetector(ipResolver ip.Resolver, databasePath string) Detector {
 	return &detector{
-		databasePath: databasePath,
+		ipResolver: ipResolver,
+		locationResolver: NewResolver(databasePath),
 	}
```

Comment:
This constructor can re-use other-one, that is:
```go
return NewDetectorWithLocationResolver(ipResolver, NewResolver(databasePath))
```

That way, only one constructor has all the low-level struct initialisation stuff.